autoload/LanguageClient.vim

```diff
@@ -1634,4 +1657,80 @@ function! LanguageClient#debugInfo(...) abort
     return LanguageClient#Call('languageClient/debugInfo', l:params, l:Callback)
 endfunction
 
+function! s:ClosePopups(...) abort
+  if s:ShouldUseFloatWindow()
+    call s:CloseFloatingHover()
+  elseif exists('*popup_clear') && s:GetVar('LanguageClient_usePopupHover', v:true)
+    call popup_clear()
+  else
+    :pclose
+  endif
+endfunction
+
+" receives the v:event from the CompleteChanged autocmd
+function! LanguageClient#handleCompleteChanged(event) abort
+  " this timer is just to stop textlock from locking our changes
+  call timer_start(0, funcref('s:ClosePopups'))
+
+  if has_key(s:timers, 'LanguageClient#handleCompleteChanged')
+    call timer_stop(s:timers['LanguageClient#handleCompleteChanged'])
+  endif
+
+  let s:timers['LanguageClient#handleCompleteChanged'] = timer_start(100, funcref('LanguageClient#showCompletionItemDocumentation', [a:event]))
```

Comment:
I used this instead of the existing `s:Debounce` because it was giving me very dubious results (it would trigger the function while holding down they key which I believe shouldn't happen with the timeout I gave it). I suppose that's due to the fact that it is more involved than using a timer, or maybe just calculating that based on `reltime` is not a reliable enough method.

I think we should probably move to this approach as a way of debouncing, as it seems more precise.